modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb

```diff
@@ -37,73 +32,34 @@ def run
 
       def default_options
         {
-          attachments: [], # an array of attachment confirmation codes
-          file_path: nil, # file path for the PDF file to be archived
-          include_json_archive: true, # include the form data as a JSON object
-          include_manifest: true, # include a CSV file containing manifest data
-          include_text_archive: true, # include the form data as a text file
-          metadata: {} # pertinent metadata for original file upload/submission
+          attachments: nil,           # The confirmation codes of any attachments which were originally submitted
+          benefits_intake_uuid: nil,  # The UUID returned from the Benefits Intake API upon original submission
+          file_path: nil,             # The local path where the submission PDF is stored
+          include_json_archive: true, # Include the form data as a JSON object
+          include_manifest: true,     # Include a CSV file containing manifest data
+          include_text_archive: true, # Include the form data as a text file
+          metadata: nil,              # Data appended to the original submission headers
+          submission: nil             # The FormSubmission object representing the original data payload submitted
         }
       end
 
+      def valid_submission_data?(data)
+        data[:submission] && data[:file_path] && data[:attachments] && data[:metadata]
+      end
+
       def process_submission_files
         write_pdf
         write_as_json_archive if include_json_archive
         write_as_text_archive if include_text_archive
         write_attachments unless attachments.empty?
         write_manifest if include_manifest
         write_metadata
+      rescue => e
```

Comment:
On line 54, if we're initializing `attachments` as `nil` then calling `empty?` on `nil` will error out.

Reply:
Good call, I went back and forth on this with hydrate_submission_data but I forgot to also account for the post-submission route.

Reply:
Fixed!